code/scripting/api/objs/subsystem.cpp

```diff
@@ -649,12 +649,21 @@ ADE_VIRTVAR(NextFireTimestamp, l_Subsystem, "number", "The next time the turret
 	if (!sso->isValid())
 		return ade_set_error(L, "f", -1.0f);
 
-	if(ADE_SETTING_VAR)
+	float currentTime = f2fl(Missiontime);
+
+	if (ADE_SETTING_VAR)
 	{
-		sso->ss->turret_next_fire_stamp = (int)(newVal * 1000);
+		float delta = newVal - currentTime;
+		if (delta < 0.0f)
+		{
+			Warning(LOCATION, "NextFireTimestamp: Specified value is in the past; setting to the current time");
+			delta = 0.0f;
+		}
```

Comment:
What's wrong with setting a time in the past?

Reply:
Well, the earliest the turret can fire is *now*, not several seconds ago.

Reply:
Sure, it may not obviously affect behavior, but it's a valid input for the value. A scripter might be interested in how long the turret has been able to fire, or want to that to be a particular value. It's a timestamp, and there's nothing inherently wrong with a timestamp in the past, right?

Reply:
It's certainly *possible*, but I don't understand why you would want to *set* the timestamp to some point in the past.  It seems to me that a negative RHS value is likely to be a bug or mistake.

Reply:
Just because we can't think of the use case doesn't mean I think we should immediately warn for it. Sure it may be a bug, but if it is a valid input, I don't think we should be complaining so loudly. In my opinion, scripting should really only warn on things which are completely invalid, and have *no possible* valid usage.

Reply:
Actually, I can't change it, because the game doesn't actually allow you to create a timestamp in the past.  All timestamps with a negative delta are treated as invalid.